### PR TITLE
Filter layout by ignored file names

### PR DESCRIPTION
**Pull request by TimFoerster**
The Layouts get filtered by the filesystem definition of ignored files.
#814

### CODE REVIEW

File: modules/cms/classes/Page.php [hunk 1]
```diff
@@ -88,8 +89,24 @@ public function getLayoutOptions()
         $layouts = Layout::listInTheme($theme, true);
         $result = [];
         $result[null] = Lang::get('cms::lang.page.no_layout');
+        $ignoreNames = FileDefinitions::get('ignoreFiles');
+
         foreach ($layouts as $layout) {
             $baseName = $layout->getBaseFileName();
+
```
Comment (LukeTowers) [inline modules/cms/classes/Page.php]: Please update this logic to mirror the logic used in the [Cms\Classes\MediaLibrary](https://github.com/octobercms/october/blob/master/modules/cms/classes/MediaLibrary.php#L515) class. We like to keep things simple in October :)

**Comment by LukeTowers** [comment]
Also this pull request has been made to the wrong branch. Please review [the contributing guidelines](https://github.com/octobercms/october/blob/master/CONTRIBUTING.md#pull-requests) as all PRs need to be made to the develop branch.

I'll fix it for you this time, but please ensure you make any future PRs to the develop branch, not the master branch.

File: modules/cms/classes/Page.php (hunk 1 above)
Comment (TimFoerster) [inline modules/cms/classes/Page.php]: The problem is that it just looks for the exact file name and not for patterns (=> dosn't resolve problems with filtering folders). However ignore patterns doesn't exists for [October\Rain\Filesystem\Definitions](https://github.com/octoberrain/filesystem/blob/master/Definitions.php#L45).
Idea 1.: Adding ignore file patterns to filesystem definitions and using it
Idea 2.: Check folder names of files and filter every folder with ignore files
What do you prefer?

Comment (LukeTowers) [inline modules/cms/classes/Page.php]: I would prefer adding ignorePatterns to the filesystem definitions to make it more consistent with the rest of October, but I'll double check with @daftspunk.

**Comment by daftspunk** [comment]
Thanks for this!

**Pull request merged**